### DISCUSSION TITLE
Connector to S3 do not work

### DISCUSSION THREAD

**Discussion by apeopleservice** [directus/directus#10739] [Archive]
After upgrade to 9.4.0 do not work connector to s3 storage (note: never (and still) do not work selection store in administration).
the connection attempt ends: "code": "INTERNAL_SERVER_ERROR"

### To Reproduce

All connection from Directus to S3 (Scaleway) will ends with error. In administration downoad/save file? From/To storage S3? Error.

### Errors Shown

```
"errors": [
{
"message": "E_UNKNOWN: An unknown error happened with the file 77dcb87f-f13e-4d52-ba60-b036dcd07587.pdf.\n\nError code: BadRequest\nOriginal stack:\nBadRequest: null\n    at Request.extractError 
(..../directus/node_modules/aws-sdk/lib/services/s3.js:690:35)\n    at Request.callListeners 
(.../directus/node_modules/aws-sdk/lib/sequential_executor.js:106:20)\n    at Request.emit 
(.../directus/node_modules/aws-sdk/lib/sequential_executor.js:78:10)\n    at Request.emit 
(.../directus/node_modules/aws-sdk/lib/request.js:686:14)\n    at Request.transition
(.../directus/node_modules/aws-sdk/lib/request.js:22:10)\n    at AcceptorStateMachine.runTo 
(.../directus/node_modules/aws-sdk/lib/state_machine.js:14:12)\n    at /.../directus/node_modules/aws-sdk/lib/state_machine.js:26:10\n    at Request.<anonymous> (/.../directus/node_modules/aws-sdk/lib/request.js:38:9)\n    at Request.<anonymous> (/.../directus/node_modules/aws-sdk/lib/request.js:688:12)\n    at Request.callListeners (.../directus/node_modules/aws-sdk/lib/sequential_executor.js:116:18)",
"extensions": {
"code": "INTERNAL_SERVER_ERROR"
}
}
]
```

**Comment by rijkvanzanten** [comment]
Heya! That `BadRequest` error comes from Scaleway directly. What configuration are you using? Does Scalaway require any special flags (like s3ForcePathStyle)

**Reply by apeopleservice** [reply]
"... special flags" - not for yesterday.

My configuration connector:

STORAGE_S3_DRIVER="s3"
STORAGE_S3_KEY="xxxxxxxxxxxxxxxxxxxxxxxxx"
STORAGE_S3_SECRET="xxxxxxx-xxxxxxx-xxxxxxxxxxxxxxxxxxx"
STORAGE_S3_BUCKET="xxxxxxxxxxxx"
STORAGE_S3_ENDPOINT="s3.pl-waw.scw.cloud"

when I tested the access via aws-cli... all is ok... all data is accessible

stressful situation

**Reply by rijkvanzanten** [reply]
[I don't think anything changed in the storage setup between v9.3.0 and v9.4.0](https://github.com/directus/directus/compare/v9.3.0...v9.4.0), so I'm not sure if / how the release  itself could've caused that 🤔 

I'm also not seeing any similar issues on instances running on AWS S3 and DigitalOcean Spaces..

Are you sure nothing else changed when you made the upgrade?

**Reply by apeopleservice** [reply]
according to your findings ... maybe two things happened in one day. I'm probably testing other stores.

thanks a lot for your response
